diff -ur orig/emacspeak-advice.el new/emacspeak-advice.el
--- orig/emacspeak-advice.el	2018-09-16 14:02:58.986020123 +0700
+++ new/emacspeak-advice.el	2018-09-16 15:25:35.783409279 +0700
@@ -899,6 +899,47 @@
   (emacspeak-auditory-icon 'button))
 
 ;;}}}
+;;{{{ shell-command:
+
+(defun emacspeak-speak-shell-command-output (&optional output)
+  (emacspeak-auditory-icon 'task-done)
+  (cond
+   ((or (stringp output) (bufferp output)) nil)
+   ((not output)
+    (let ((buffer (get-buffer "*Shell Command Output*")))
+      (when buffer
+        (with-current-buffer buffer
+          (if (= (point-min) (point-max))
+              (dtk-speak (current-message))
+            (emacspeak-speak-region (point-min) (min (point-max) 8192))
+            (when (> (point-max) 8192)
+              (emacspeak-queue-auditory-icon 'ellipses)))))))
+   (:else
+    (if (= (point) (mark 'force))
+        (dtk-speak "Empty output")
+      (emacspeak-speak-region (point) (min (mark 'force) (+ (point) 8192)))
+      (when (> (mark 'force) (+ (point) 8192))
+        (emacspeak-queue-auditory-icon 'ellipses))))))
+
+(defadvice shell-command (around emacspeak activate)
+  (let ((emacspeak-speak-messages nil))
+    ad-do-it)
+  (when (called-interactively-p 'interactive)
+    (emacspeak-speak-shell-command-output (ad-get-arg 1))))
+
+(defadvice shell-command-on-region (around emacspeak activate)
+  (let ((emacspeak-speak-messages nil))
+    ad-do-it)
+  (when (called-interactively-p 'interactive)
+    (emacspeak-speak-shell-command-output (ad-get-arg 3))))
+
+(defadvice dired-do-shell-command (around emacspeak activate)
+  (let ((emacspeak-speak-messages nil))
+    ad-do-it)
+  (when (called-interactively-p 'interactive)
+    (emacspeak-speak-shell-command-output)))
+
+;;}}}
 ;;{{{ Advice comint:
 
 (defadvice comint-magic-space (around emacspeak pre act comp)
@@ -959,7 +1000,7 @@
 (add-hook 'shell-mode-hook 'emacspeak-pronounce-refresh-pronunciations)
 
 (loop for f in
-      '(shell-command shell-dirstack-message)
+      '(shell-dirstack-message)
       do
       (eval
        `(defadvice ,f (around emacspeak pre act comp)
